homeassistant/components/evohome/coordinator.py

```diff
@@ -5,85 +5,94 @@
+        self.loc: evo.Location = None  # type: ignore[assignment]
+
+        self.loc_utc_offset: timedelta = None  # type: ignore[assignment]
+        self.tcs: evo.ControlSystem = None  # type: ignore[assignment]
 
-    def __init__(
-        self,
-        hass: HomeAssistant,
-        client: evo.EvohomeClient,
-        client_v1: ev1.EvohomeClient | None,
-        store: Store[dict[str, Any]],
-        params: ConfigType,
-    ) -> None:
-        """Initialize the evohome client and its data structure."""
-        self.hass = hass
-        self.client = client
-        self.client_v1 = client_v1
-        self._store = store
-        self.params = params
-
-        loc_idx = params[CONF_LOCATION_IDX]
-        self._location: evo.Location = client.locations[loc_idx]
-
-        assert isinstance(client.installation_info, list)  # mypy
-
-        self.config = client.installation_info[loc_idx][GWS][0][TCS][0]
-        self.tcs: evo.ControlSystem = self._location._gateways[0]._control_systems[0]  # noqa: SLF001
-        self.loc_utc_offset = timedelta(minutes=self._location.timeZone[UTC_OFFSET])
         self.temps: dict[str, float | None] = {}
 
-    async def save_auth_tokens(self) -> None:
-        """Save access tokens and session IDs to the store for later use."""
-        # evohomeasync2 uses naive/local datetimes
-        access_token_expires = dt_local_to_aware(
-            self.client.access_token_expires  # type: ignore[arg-type]
-        )
+    def validate_location(self, loc_idx: int) -> bool:
+        """Get the default TCS of the specified location."""
 
-        app_storage: dict[str, Any] = {
-            CONF_USERNAME: self.client.username,
-            REFRESH_TOKEN: self.client.refresh_token,
-            ACCESS_TOKEN: self.client.access_token,
-            ACCESS_TOKEN_EXPIRES: access_token_expires.isoformat(),
-        }
+        self.loc_idx = loc_idx
 
-        if self.client_v1:
-            app_storage[USER_DATA] = {
-                SZ_SESSION_ID: self.client_v1.broker.session_id,
-            }  # this is the schema for STORAGE_VER == 1
-        else:
-            app_storage[USER_DATA] = {}
+        assert isinstance(self.client.installation_info, list)  # mypy
 
-        await self._store.async_save(app_storage)
+        try:
+            loc_config = self.client.installation_info[loc_idx]
+        except IndexError:
+            _LOGGER.error(
+                (
+                    "Config error: '%s' = %s, but the valid range is 0-%s. "
+                    "Unable to continue. Fix any configuration errors and restart HA"
+                ),
+                CONF_LOCATION_IDX,
+                loc_idx,
+                len(self.client.installation_info) - 1,
+            )
+            return False
+
+        self.loc = self.client.locations[loc_idx]
+        self.loc_utc_offset = timedelta(minutes=self.loc.timeZone[UTC_OFFSET])
+        self.tcs = self.loc._gateways[0]._control_systems[0]  # noqa: SLF001
```

Comment:
Why did you make it private in the first place if access is needed in other places?

Reply:
I never 'made' it private - it was like that when I found it. 

I converted the non-async of the client library, **evohome-client** into **evohome-async**.

I have tried to maintain the same namespace (same attrs, methods) so people could easily convert from non-async to async, and I have been limited by the original design decisions.

Reply:
I will go back and make them public, and bump the client - that will be a pre-requisite for this PR.

Reply:
Perfectly fine, you are in control of both sides so I think the implementation should reflect that and thus the # noqa should be superfluous